contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -15,8 +15,10 @@ type getarg >/dev/null 2>&1 || . "$dracutlib"
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
 
 rootfstype=zfs
-if echo "${rootflags}" | grep -q zfsutil ; then
+if echo "${rootflags}" | grep -Eq '^zfsutil$|^zfsutil,|,zfsutil$|,zfsutil,' ; then
     true
+elif test -n "${rootflags}" ; then
+    rootflags="zfsutil,${rootflags}"
 else
     rootflags=zfsutil
 fi
```

Comment:
I am inclined to believe you, but, have you verified that there will be no sysroot.mount in all supported distributions and versions?  At the very least, a test for this behaviour would include Fedora 20 and 23, and CentOS 7 and 6, verifying that systemctl show sysroot.mount's LoadState is empty (or, alternatively, that they boot).


Where should that snippet go?